### ISSUE TITLE
Response not coming through for every request with intercept

### ISSUE THREAD

**Issue by Tom5om** [cypress-io/cypress#14522]
Hi There,

I was upgrading cypress in our application and migrating to intercept. 
It seems to work intermittently. 

**OLD**
```js
cy.server()
        .route('post', '**/landlord')
        .as('saveLandlord');

cy.contains('button', 'Next').click();

cy.wait('@saveLandlord').should('have.property', 'status', 200);
```

**New**

```js
cy.intercept('POST', '**/landlord**').as('saveLandlord');

cy.contains('button', 'Next').click();

cy.wait('@saveLandlord')
        .its('response.statusCode')
        .should('eq', 200);
```

I have tried to log the response in between and the response is just not passed in.

This is the error I get, and you can see that it did not yield a response on the right:
![image](https://user-images.githubusercontent.com/4668890/104411681-f0fa1300-55be-11eb-9106-02bf0aa3824f.png)

Here is the response from the console:
![image](https://user-images.githubusercontent.com/4668890/104411723-05d6a680-55bf-11eb-876c-273a9e450a96.png)

The same code works for some other requests, I can't really see why this doesn't work for this instance.

**Comment by jennifer-shehane** [comment]
Are you using the latest version of Cypress? Please update to the [current version of Cypress](https://on.cypress.io/changelog) if not - as we've fixed issues that look similar to this in more recent versions.

**Comment by LoganFievez** [comment]
I downloaded the last version of cypress today and I encountered the same issue

**Comment by MistertempDev** [comment]
We're having the **exact** same problem also! We're using Cypress 6.2.1 on Mac (and Windows).

**Comment by bahmutov** [comment]
Can someone provide a reproducible example for us to see this behavior? We have `cy.intercept` recipe in https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/stubbing-spying__intercept that you can use as the basis for the reproduction.

**Comment by jaiswalkpraveen** [comment]
Can we connect right away? I'm facing same issue but not sure what exactly you are looking for?
   ``` cy.intercept('POST', '/graphql', (req) => {
        if (req.body.query.includes('get_user_credits')) {
            req.alias = 'gqlUserCreditsQuery'
        }
    })
     cy.get(':nth-child(1) > .MuiBottomNavigationAction-wrapper').contains('Courses').should('be.visible').click()
      cy.wait('@gqlUserCreditsQuery').should((API) => {
            console.log(API)
       })
       ```

**Comment by jaiswalkpraveen** [comment]
@bahmutov I'm intercepting graphQL request and verifying it's response. Let me know if we can connect quickly.
![image](https://user-images.githubusercontent.com/6623471/104474653-3da41500-55e4-11eb-9383-d205317c270e.png)

**Comment by bahmutov** [comment]
Hi @jaiswalkpraveen can you not create a reproducible example we can run?

**Comment by jaiswalkpraveen** [comment]
Give me 5 minutes, on it.

**Comment by jaiswalkpraveen** [comment]
What is missing in above snaps exactly? what else you need ?
![Uploading image.png…]()

**Comment by bahmutov** [comment]
We cannot run the image or just the test without the app, right

**Comment by jaiswalkpraveen** [comment]
@bahmutov Got it. Unfortunately, my apps needs credentials and I can't share it.

**Comment by jaiswalkpraveen** [comment]
@bahmutov Just tried with external website, it's working intermittently. Please find script below.
Let me try to brief it. 
1. Launch website
2. Click on any webpage by clicking pipeline button which is like 'waiting for page to load' for more than 5-10 sec
3. Then I'm getting this error

it("validate same gql response ", () => {
    cy.intercept('POST', '_graphql/v1', (req) => {
        if (req.body.query.includes('Index_ViewerRelayQL')) {
            req.alias = 'IndexQuery'
        }

    })
    cy.visit('https://buildkite.com/')
    cy.contains('Start Trial').click()
    cy.get('.input').eq(0).type('Kumar')
    cy.get('.input').eq(1).type('Lavasa')
    cy.get('.input').eq(2).type('Praveen@mira.io')
    cy.get('.input').eq(3).type('Cypress*&^%')
    cy.contains('Get Started').click()
    cy.wait('@IndexQuery').then((APIResponse1) => {
        console.log(APIResponse1)
    })
    cy.contains('Pipelines').click()
    cy.wait('@IndexQuery').then((APIResponse2) => {
        console.log(APIResponse2)
    })
})

**Comment by Tom5om** [comment]
Hi There, yep I use version 6.2.1, apologies for not mentioning that.

does the example above suffice? I can't use our own application as an example, and like I said this doesn't happen for all requests in my application, but it always happen for a whole bunch of them. I can't see what the difference is

**Comment by Tom5om** [comment]
Hi @bahmutov I have been able to create an example where this happens:

```
it('Login page shows', () => {
            cy.visit('app.staging.flkitover.com');
            cy.clearCookies();
            cy.location('pathname').should('equal', '/');
            cy.get('h1').contains('Log In');

            cy.intercept('POST', '/login').as('login');
            // enter valid username and password
            cy.get('[name=email]').type('something@something.com');
            cy.get('[name=password]').type('Test1234');
            cy.contains('button', 'Log in').click();

            cy.wait('@login')
                .its('response.statusCode')
                .should('eq', 401);

            cy.intercept('POST', '/login').as('login2');
            // enter valid username and password
            cy.get('[name=password]').type('Test1234');
            cy.contains('button', 'Log in').click();

            cy.wait('@login2')
                .its('response.statusCode')
                .should('eq', 401);
        });
```

**Comment by bahmutov** [comment]
@Tom5om great reproduction, thank you so much. I can see it in reproduction https://github.com/cypress-io/cypress-test-tiny/tree/response

There are two spy intercepts.

First we define `login` spy intercept and test it

So the Ajax call fails

<img width="1046" alt="Screen Shot 2021-01-14 at 12 25 38 PM" src="https://user-images.githubusercontent.com/2212006/104626350-cdd56f00-5663-11eb-90b6-661145d5258c.png">

If we ask the first spy, it has the right status code and response

<img width="1027" alt="Screen Shot 2021-01-14 at 12 28 58 PM" src="https://user-images.githubusercontent.com/2212006/104627022-856a8100-5664-11eb-9844-6ddcf9b16632.png">

Then we define the _second_ spy intercept for the same route

and make the Ajax call again

Since these are spies, the first spy intercept is called again (so it has count 2), and the second spy intercept is only called once

<img width="440" alt="Screen Shot 2021-01-14 at 12 29 33 PM" src="https://user-images.githubusercontent.com/2212006/104627116-9dda9b80-5664-11eb-9db3-4e0f11d1d90c.png">

The error response from the server goes through the first intercept, and when we wait for second intercept it gets no `response` object

<img width="1040" alt="Screen Shot 2021-01-14 at 12 25 52 PM" src="https://user-images.githubusercontent.com/2212006/104627214-bcd92d80-5664-11eb-9ab4-c78e50ab4e44.png">

Seems we need to make sure spy intercepts pass everything correctly down the line if the network call had an error

**Comment by Tom5om** [comment]
@bahmutov @flotwig just want to clarify, that this is how I was able to reproduce it.

I couldn't give you examples, but it happens as well when I intercept different routes (but multiple) and also when there are 200 statuscode responses

**Comment by DominikPalo** [comment]
I have the same issue:

We have replaced the old
```
    cy
      .route({
        method: 'POST',
        url: '**/oauth/token?grant_type=password&**'
      }).as('authentication');
```

with

```
    cy
       .intercept('POST', '**/oauth/token?grant_type=password&**')
       .as('authentication');
``` 


and now the yielded intercept object does not contain the response.

```
    cy
      .log('authentication request successful')
      .wait('@authentication')
      .then((auth) => {
        const accessToken = auth.response.body['access_token'];
     });
```

fails with
`Cannot read property 'body' of undefined`

because the actual intercept object contains only the request object:
```
{
  id: "interceptedRequest82"
  log: {get: ƒ, unset: ƒ, invoke: ƒ, toJSON: ƒ, set: ƒ, …}
  request: {headers: {…}, body: {…}, url: "...", method: "POST", httpVersion: "1.1"}
  requestWaited: true
  responseWaited: true
  routeHandlerId: "1611048999315-1637"
  state: "Complete"
  __proto__: Object
}
```

**Comment by kpurdon** [comment]
Does anybody have a workaround on 6+ for this issue?

**Comment by Tom5om** [comment]
@kpurdon the workaround is to use the old way: 

```
cy.server()
        .route('post', '**/landlord')
        .as('saveLandlord');

cy.contains('button', 'Next').click();

cy.wait('@saveLandlord').should('have.property', 'status', 200);
```

**Comment by danisal** [comment]
Facing the same issue in version 6.4.0

**Comment by CypressHarry** [comment]
I believe this is a duplicate of #14205

**Comment by sainthkh** [comment]
I partially solved the bug. Current problem is that it breaks other features. I'm trying to find the solution.

**Comment by jprust** [comment]
@flotwig would it be possible to review Sainthkh's PR and include this fix in the next release?

**Comment by flotwig** [comment]
@jprust the fix will be included with 7.0.0.
